stories/core/ColorPicker.stories.tsx

```diff
@@ -0,0 +1,132 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { Story, Meta } from '@storybook/react';
+import React, { useState } from 'react';
+import {
+  ColorPicker,
+  ColorPickerProps,
+  IconButton,
+  Tooltip,
+} from '../../src/core';
+import { action } from '@storybook/addon-actions';
+import ColorSwatch from '../../src/core/ColorPicker/ColorSwatch';
+import { DefaultColors } from '../../src/core/ColorPicker/ColorPicker';
+
+export default {
+  component: ColorPicker,
+  argTypes: {
+    className: { control: { disable: true } },
+    style: { control: { disable: true } },
+    children: { control: { disable: true } },
+  },
+  title: 'Core/ColorPicker',
+} as Meta<ColorPickerProps>;
+
+export const Basic: Story<ColorPickerProps> = (args) => {
+  const [activeIndex, setActiveIndex] = useState(5);
+
+  const [opened, setOpened] = useState(false);
+
+  const [currentColorName, setCurrentColorName] = useState(
+    DefaultColors[activeIndex].name,
+  );
+  const [currentColorValue, setCurrentColorValue] = useState(
+    DefaultColors[activeIndex].color,
+  );
+
+  const onClickColor = (index: number) => {
+    action(`Clicked color ${DefaultColors[index].color}`)();
+    setActiveIndex(index);
+    setCurrentColorName(DefaultColors[index].name);
+    setCurrentColorValue(DefaultColors[index].color);
+  };
+
+  return (
+    <>
+      <IconButton onClick={() => setOpened(!opened)}>
+        <span
+          style={{ backgroundColor: currentColorValue, border: '1px solid' }}
+        />
+      </IconButton>
+      <span style={{ marginLeft: 16 }}>{currentColorName}</span>
+      {opened && (
+        <div style={{ marginTop: 4 }}>
+          <ColorPicker {...args}>
+            {DefaultColors.map((color, index) => {
+              const onClick = () => {
+                onClickColor(index);
+              };
+              return (
+                <ColorSwatch
+                  key={index + color.color}
+                  color={color.color}
+                  onClick={onClick}
+                  isActive={color.color === currentColorValue}
+                />
+              );
+            })}
+          </ColorPicker>
+        </div>
+      )}
+    </>
+  );
+};
+
+Basic.args = {};
+
+export const WithTooltip: Story<ColorPickerProps> = (args) => {
+  const [opened, setOpened] = useState(false);
+
+  const [activeIndex, setCurrentActiveIndex] = React.useState(10);
+  const [currentColorName, setCurrentColorName] = useState(
+    DefaultColors[activeIndex].name,
+  );
+  const [currentColorValue, setCurrentColorValue] = useState(
+    DefaultColors[activeIndex].color,
+  );
```

Comment:
So we have 3 variables for managing the same state? I think we can show a cleaner example.

Reply:
Ok I cleaned it up